packages/vue-app/template/components/nuxt-link.client.js

```diff
@@ -0,0 +1,85 @@
+<%= isTest ? '// @vue/component' : '' %>
+import Vue from 'vue'
+
+const requestIdleCallback = window.requestIdleCallback ||
+  function (cb) {
+    const start = Date.now();
+    return setTimeout(function () {
+      cb({
+        didTimeout: false,
+        timeRemaining: function () {
+          return Math.max(0, 50 - (Date.now() - start));
+        },
+      });
+    }, 1);
+  }
+const observer = window.IntersectionObserver && new window.IntersectionObserver(entries => {
+  entries.forEach(entry => {
+    if (!entry.isIntersecting) {
+      return
+    }
+    const link = entry.target
+
+    observer.unobserve(link)
+    link.__prefetch()
+  })
+})
+
+export default {
+  extends: Vue.component('RouterLink'),
+  name: 'NuxtLink',
+  props: {
+    noPrefetch: {
+      type: Boolean,
+      default: false
+    }
+  },
+  mounted() {
+    if (!this.noPrefetch) {
+      requestIdleCallback(this.observe, { timeout: 2e3 })
+    }
+  },
+  beforeDestroy() {
+    if (this.__observed) {
+      observer.unobserve(this.$el)
+      delete this.$el.__prefetch
+    }
+  },
+  methods: {
+    observe() {
+      // If no IntersectionObserver, prefetch directly
```

Comment:
Well, it will make XHR requests that will be handled by the browser, I just need to add a function to handle poor connection to not do such requests :)

Reply:
Still, they are normal XHR requests with the same priority as normal API requests. We can't also at least with current webpack API prioritize its XHR requests. So at least we can try to reduce them.

Reply:
I removed the call finally, the will use a polyfill (added to the docs) if they want to support this feature :)